crates/nu-command/src/system/benchmark.rs

```diff
@@ -41,12 +45,27 @@ impl Command for Benchmark {
         let redirect_stderr = call.redirect_stderr;
 
         let mut stack = stack.captures_to_stack(&capture_block.captures);
+
+        // In order to provide the pipeline as a positional, it must be converted into a value.
+        // But because pipelines do not have Clone, this one has to be cloned as a value
+        // and then converted back into a pipeline for eval_block().
+        // So, the metadata must be saved here and restored at that point.
+        let input_metadata = input.metadata();
+        let input_val = input.into_value(call.head);
+
+        if let Some(var) = block.signature.get_positional(0) {
+            if let Some(var_id) = &var.var_id {
+                stack.add_var(*var_id, input_val.clone());
+            }
+        }
+
+        // Get the start time after all other computation has been done.
         let start_time = Instant::now();
         eval_block(
             engine_state,
             &mut stack,
             block,
-            PipelineData::empty(),
+            input_val.into_pipeline_data_with_metadata(input_metadata),
```

Comment:
Yeah that creates a `PipelineData::Value` and is not specialized for `Value::List` or `ListStream`

Reply:
This isn't supposed to work like an iterator command like `each`. For `[1 2 3] | benchmark { $in }`, `$in` is `[1 2 3]`, not `1`, `2` and `3` respectively. That means all stream inputs must be collected in order to be passed in as a value  (which is what `input.into_value()` does) and thus I believe producing `PipelineData::Value` is semantically correct.